PolicyDecisionPoint/src/main/java/it/cnr/iit/ucs/pdp/PolicyDecisionPoint.java

```diff
@@ -231,25 +213,13 @@ private ResponseCtx evaluate( EvaluationCtx context, PolicyFinder policyFinder )
                 return new ResponseCtx( results, XACMLConstants.XACML_VERSION_3_0 );
             }
         } else {
-            // this is special case that specific to XACML3 request
-            if( context instanceof XACML3EvaluationCtx
-                    && ( (XACML3EvaluationCtx) context ).isMultipleAttributes() ) {
-                ArrayList<String> code = new ArrayList<>();
-                code.add( Status.STATUS_SYNTAX_ERROR );
-                Status status = new Status( code,
-                    "PDP does not supports multiple decision profile. "
-                            + "Multiple AttributesType elements with the same Category can be existed" );
-                return new ResponseCtx( ResultFactory.getFactory()
-                    .getResult( AbstractResult.DECISION_INDETERMINATE, status, context ) );
-            } else if( context instanceof XACML3EvaluationCtx
-                    && ( (RequestCtx) context.getRequestCtx() ).isCombinedDecision() ) {
-                List<String> code = new ArrayList<>();
-                code.add( Status.STATUS_PROCESSING_ERROR );
-                Status status = new Status( code,
-                    "PDP does not supports multiple decision profile. "
-                            + "Multiple decision is not existed to combine them" );
-                return new ResponseCtx( ResultFactory.getFactory()
-                    .getResult( AbstractResult.DECISION_INDETERMINATE, status, context ) );
+            // this is a special case that specific to XACML3 request
+            if( context instanceof XACML3EvaluationCtx && ( (XACML3EvaluationCtx) context ).isMultipleAttributes() ) {
+                return getResponseCtxFor( AbstractResult.DECISION_INDETERMINATE, Status.STATUS_SYNTAX_ERROR,
+                    "Usupported multiple decision profile. Multiple AttributesType same Category can exist", context );
+            } else if( context instanceof XACML3EvaluationCtx && ( (RequestCtx) context.getRequestCtx() ).isCombinedDecision() ) {
+                return getResponseCtxFor( AbstractResult.DECISION_INDETERMINATE, Status.STATUS_PROCESSING_ERROR,
+                    "Unsupported multiple decision profile. Is's not possible to combine them multiple decisions", context );
             } else {
```

Comment:
I think we can reove the "else" and above in line 220

Reply:
yes!